docs/kubernetes/cross-cluster.md

```diff
@@ -0,0 +1,229 @@
+
+1. Create `StorageClass` in all the clusters. In this example, ReclaimPolicy: `Retain` is used
+
+    If you haven't created a `StorageClass` yet, create one first:
+
+    ```console
+    $ kubectl apply -f ./examples/kubernetes/cross-cluster/sc.yaml
+    ```
+
+    If a non-default network is used for the filestore instance, provide a network paramter to the storage class.
+
+    ```yaml
+    apiVersion: storage.k8s.io/v1
+    kind: StorageClass
+    metadata:
+      name: csi-filestore
+    provisioner: filestore.csi.storage.gke.io
+    reclaimPolicy: Retain
+    parameters:
+      network: <network name> # Change this network as per the deployment
+    volumeBindingMode: WaitForFirstConsumer
+    allowVolumeExpansion: true
+    ```
+2. In `clusterA`, create the deployment as follows. 
+
+    ```console
+    $ kubectl apply -f ./examples/kubernetes/cross-cluster/demo-deployment-clusterA.yaml
+    ```
+    
+    As part of the worklow to bringup the deployment, the CSI filestore driver in Cluster A would dynamically create the filestore instance and map to the PersistentVolume used by the deployment.
+
+    ```console
+    $ kubectl get pvc
+    NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+    test-pvc-fs   Bound    pvc-88959393-01c7-415f-aac4-a974c45f7ec7   1Ti        RWX            csi-filestore   4m17s
+    ```
+
+    The PersistentVolume is created with a ReclaimPolicy `Retain`.
+    ```console
+    $ k get pv pvc-88959393-01c7-415f-aac4-a974c45f7ec7
+    NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                 STORAGECLASS    REASON   AGE
+    pvc-88959393-01c7-415f-aac4-a974c45f7ec7   1Ti        RWX            Retain           Bound    default/test-pvc-fs   csi-filestore            2m37s
+    ```
+
+    From `clusterA`, ensure the pods are up and running
+    ```console
+    $ kubectl get pod
+    NAMESPACE                  NAME                                                      READY   STATUS    RESTARTS   AGE
+    default                    web-server-deployment-cluster-a-6fc54d4776-7lzkb          1/1     Running   0          3m3s
+    default                    web-server-deployment-cluster-a-6fc54d4776-nspm7          1/1     Running   0          3m3s
+    default                    web-server-deployment-cluster-a-6fc54d4776-qdttl          1/1     Running   0          3m3s
+    ```
+3. Capture the filestore instance details i.e the filestore instance name, share name from `VolumeHandle` and the filestore instance IP from `VolumeAttributes`.
```

Comment:
Consider utilizing the json path to get the values from a kubectl command utilizing "-o jsonpath", eg.
```
kubectl get pv pvc-018ded1d-bec3-40c1-a98b-28c4c24d0561 -o "jsonpath={.spec.csi.volumeAttributes.ip}"
```

Reply:
Done